src/fiatExchanges/quotes/FiatConnectQuote.ts

```diff
@@ -152,12 +208,37 @@ export default class FiatConnectQuote extends NormalizedQuote {
     return this.quoteResponseKycSchema?.kycSchema
   }
 
-  // TODO: Dynamically generate time estimation strings
-  getTimeEstimation(): SettlementTime {
-    // payment method can only be bank or fc mobile money
-    return this.getPaymentMethod() === PaymentMethod.Bank
-      ? SettlementTime.ONE_TO_THREE_DAYS
-      : SettlementTime.LESS_THAN_24_HOURS
+  getTimeEstimation(): SettlementEstimation {
+    const fiatAccountInfo = this.quote.fiatAccount[this.getFiatAccountType()]
+
+    const lowerBound = fiatAccountInfo?.settlementTimeLowerBound
+      ? parseInt(fiatAccountInfo?.settlementTimeLowerBound)
+      : 0
+    const upperBound = fiatAccountInfo?.settlementTimeUpperBound
+      ? parseInt(fiatAccountInfo?.settlementTimeUpperBound)
+      : 0
+    if (
+      !fiatAccountInfo?.settlementTimeUpperBound ||
```

Comment:
I would imagine that after setting the `upperBound` and `lowerBound` variables, we should avoid directly accessing `fiatAccountInfo.settlmentTimeUpperBound` again. Can we just roll this first check into the `upperBound < 0` check by doing `upperBound <= 0` instead? It should be OK to keep the `lowerBound < 0` check as-is, since we're still capable of giving a non-default estimation when the provider doesn't return a lower bound.

Reply:
Good call, updated the check